smithy-rules-engine/src/main/java/software/amazon/smithy/rulesengine/traits/EndpointTestsTraitValidator.java

```diff
@@ -36,7 +39,28 @@ public List<ValidationEvent> validate(Model model) {
                 operationNameMap.put(operationShape.getId().getName(), operationShape);
             }
 
+            // Precompute the built-ins and their default states, as this will
+            // be used frequently in downstream validation.
+            List<Parameter> builtInParamsWithDefaults = new ArrayList<>();
+            List<Parameter> builtInParamsWithoutDefaults = new ArrayList<>();
+            EndpointRuleSet ruleSet = serviceShape.expectTrait(EndpointRuleSetTrait.class).getEndpointRuleSet();
+            for (Parameter parameter : ruleSet.getParameters()) {
+                if (parameter.isBuiltIn()) {
+                    if (parameter.getDefault().isPresent()) {
+                        builtInParamsWithDefaults.add(parameter);
+                    } else {
+                        builtInParamsWithoutDefaults.add(parameter);
+                    }
+                }
+            }
+
             for (EndpointTestCase testCase : serviceShape.expectTrait(EndpointTestsTrait.class).getTestCases()) {
+                // If values for built-in parameters don't match the default, they MUST
+                // be specified in the operation inputs. Precompute the ones that don't match
+                // and capture their value.
+                Map<Parameter, Node> mismatchedBuiltInParams =
+                        getMismatchedBuiltInParams(builtInParamsWithDefaults, testCase);
```

Comment:
"Mismatched" doesn't quite feel like the right description - to be it sounds like these are a invalid (esp in the context of a validator) - maybe something like `builtInsWithNonDefaultValues`